docs/TEMPORARY_DIRECTORIES.md

```diff
@@ -0,0 +1,201 @@
+   sub-directory. This is particularly interesting when operating on more than a
+   single file, or on file nodes that are not plain regular files, for example
+   when extracting a tarball to a temporary directory. The ageing algorithm
+   will skip all directories (and everything below them) that are locked
+   through a BSD file lock. As BSD file locks are automatically released when
+   the file descriptor they are taken on is closed, and all file descriptors
+   opened by a process are implicitly closed when it exits this is a robust
+   mechanism that ensures all generated temporary files are subject to ageing
+   when the program that owns them abnormally dies, but not while it is still
+   running. This is a particularly interesting option when decompressing
+   tarballs that contain files with old modification/access times, as extracted
+   files are otherwise immediately candidates for deletion by the ageing
+   algorithm. The [`flock`](http://man7.org/linux/man-pages/man1/flock.1.html)
+   tool of the `util-linux` packages makes this concept available to shell
+   scripts. Note that `systemd-tmpfiles` only checks for BSD file locks on
+   directories, locks on other types of file nodes (including regular files)
+   are not considered.
+
+4. Keep the access time of all temporary files created current. In regular
+   intervals, use `utimensat()` or a related call to update the access time
+   ("atime") of all files that shall be kept around. Since the ageing algorithm
+   looks at the access time of files when deciding whether to delete them, it's
+   sufficient to update their access times in sufficiently frequent intervals to
+   ensure the files are not deleted. Since most applications (and tools such as
+   `ls`) primarily care for the modification time (rather than the access time)
+   using the access time for this purpose should be acceptable.
+
+5. Set the "sticky" bit on regular files. The ageing logic skips deletion of
+   all regular files that have the sticky bit (`chmod +t`) set. This is
+   honoured for regular files only however, and has no effect on directories as
+   the sticky bit has a different meaning for them.
+
+6. Don't use `/tmp/` or `/var/tmp/`, but use you own sub-directory under `/run/`
+   or `$XDG_RUNTIME_DIRECTORY` (the former if privileged, the latter if
+   unprivileged), or `/var/lib/` and `~/.config/` (similar, but with
+   persistency and suitable for larger data). The two temporary directories
+   `/tmp/` and `/var/tmp/` come with the implicit clean-up semantics described
+   above. When this is not desired, it's possible to create private per-package
+   runtime or state directories, and place all temporary files there. However,
+   do note that this means opting out of any kind of automatic clean-up, and it
+   is hence particularly essential that the program cleans up generated files
+   in these directories when they are no longer needed, in particular when the
+   program dies abnormally.
```

Comment:
Hmm, I would be inclined to drop this paragraph, or at least add a big caveat that this only applies to permanent or semi-permanent stuff. We shouldn't be telling people to use those other directories just to avoid cleanup.

Reply:
I'll add some explanation about this.

(btw, this recommendation is in context of "dracut": it needs some temporary space, and for "dracut" a directory below /run might be an option since it's usually run in a single instance, where each run can in the worst case clean up the left-over artifacts from the previous run)